### PR TITLE
add document for Download

### PR DESCRIPTION
**Pull request by zydxhs**
This PR is the document for PR #599.

### CODE REVIEW

File: advanced-usage.md [hunk 1]
```diff
@@ -540,6 +540,58 @@ std::cout << r.text << std::endl;
...
+
+### Download With Callback
+
+When downloading a small file, you might want to allocate enough memory to hold the data you read before starting the download. This is where 'GetDownloadFileLength()' comes in.
+
+{% raw %}
+```c++
+struct File
+{
+    void *  file_buf;  // file data will be save to
```
Comment (COM8) [inline advanced-usage.md:567]:
```suggestion
    void*  file_buf;  // file data will be save to
```

File: advanced-usage.md [hunk 2]
```diff
@@ -540,6 +540,58 @@ std::cout << r.text << std::endl;
...
+    int64_t read_len;  // file bytes
+};
+bool write_data(std::string data, intptr_t userdata)
+{
+    File *pf = (File *)userdata;
```
Comment (COM8) [inline advanced-usage.md:572]:
```suggestion
    File* pf = static_cast<File*>(userdata);
```

**Review by COM8 — CHANGES_REQUESTED** [review]
Thanks for taking the time to add the documentation here.
Only two small things to fix and we are ready to go.

**Comment by zydxhs** [comment]
The pointer's position fixed.

**Review by COM8 — APPROVED** [review]
Thanks!

**Pull request merged**